lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -0,0 +1,247 @@
+      list.flatten
+    end
+
+    # This method will generate a batched string if a list is passed as
+    # argument
+    # Input would be as [1,2,3,4,5,10,11,12,7,30,100,31,32]
+    # output will be 1-5,10-12,7,30,100,31-32
+    def self.bd_list_to_string(bd_list)
+      farray = bd_list.compact
+      lranges = []
+      unless farray.empty?
+        left = bd_list.first
+        right = nil
+        farray.each do |aelem|
+          if right && aelem != right.succ
+            if left == right
+              lranges << left
+            else
+              lranges << Range.new(left, right)
+            end
+            left = aelem
+          end
+          right = aelem
+        end
+        if left == right
+          lranges << left
+        else
+          lranges << Range.new(left, right)
+        end
+      end
+      lranges.to_s.gsub('..', '-').delete('[').delete(']').delete(' ')
+    end
+
+    def cli_error_check(result)
+      # The NXOS bridge-domain cli does not raise an exception in some
+      # conditions and instead just displays a STDOUT error message;
+      # thus NXAPI does not detect the failure and we must catch it by
+      # inspecting the "body" hash entry returned by NXAPI. This
+      # bridge-domain cli behavior is unlikely to change.
+      fail result[2]['body'] if
+        result[2].is_a?(Hash) &&
+        /(ERROR:|Warning:)/.match(result[2]['body'].to_s)
+
+      # Some test environments get result[2] as a string instead of a hash
+      fail result[2] if
+        result[2].is_a?(String) &&
+        /(ERROR:|Warning:)/.match(result[2])
+    end
+
+    def create
+      result = config_set('bridge_domain', 'create', @bd_ids, @bd_ids)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def destroy
+      result = config_set('bridge_domain', 'destroy', @bd_ids, @bd_ids)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def self.allbds
+      hash = {}
+      bd_list = config_get('bridge_domain', 'all_bds')
+      return hash if bd_list.nil?
+
+      bd_list.each do |id|
+        hash[id] = BridgeDomain.new(id, false)
+      end
+      hash
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    # Bridge-Domain Shutdown case
+    def shutdown
+      result = config_get('bridge_domain', 'shutdown', @bd_ids)
+      # Valid result is either: "active"(aka no shutdown) or "shutdown"
+      result[/DOWN/] ? true : false
+    end
+
+    def shutdown=(val)
+      no_cmd = (val) ? '' : 'no'
+      result = config_set('bridge_domain', 'shutdown', @bd_ids, no_cmd)
+      cli_error_check(result)
+    rescue CliError => e
+      raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
```

Comment:
This is okay but consider using a helper method to do this raise instead of replicating it on every setter.

```
rescue CliError => e
  bd_raise(e)
end

def bd_raise(e)
  raise "[bridge-domain #{@bd_ids}] '#{e.command}' : #{e.clierror}"
end